src/test/java/com/sap/sgs/phosphor/fosstars/model/score/WeightedCompositeScoreTest.java

```diff
@@ -25,10 +25,18 @@
 
 public class WeightedCompositeScoreTest {
 
+  private static final double PRECISION = 0.001;
```

Comment:
Other question: 
- Can this precision vary with time? Maybe after the inclusion of new features or the addition of new scores?
- Should we think about maybe having a range for this [0.01, 0.001].

Reply:
The precision depends on values which are used in the test. At the moment, the test uses `x.yz` doubles. The current precision should be enough to test those doubles. If the test starts using longer doubles, then the precision will need to be updated. I am not sure why we need a range for precisin.

Reply:
I understood it wrong then, I thought the value is being compared with precision in difference. That is the expectedValue cannot be more or less than the actualValue by 0.001. Hence, asked if it makes sense to give a range or precision?

Reply:
> That is the expectedValue cannot be more or less than the actualValue by 0.001.

Yes, that's correct.

> Hence, asked if it makes sense to give a range or precision?

I am not sure how a range can help here. At the moment, doubles are compared like this:

```java
double actual = ...
double expected ...
assertEquals(expected, actual, PRECISION);
```

See also https://junit.org/junit4/javadoc/4.12/org/junit/Assert.html#assertEquals(double,%20double,%20double)

Such a check ends up to something like this:

```java
if (abs(actual - expected) < PRECISION) {
    // throw an exception
}
```

I am not sure how a range of precision can make it better. Or, am I missing something?

Reply:
There is also `assertScore(Interval, ...)` method which checks that a score value belongs to a specific range. I think we need to use `assertEquals(double, double, double)` when we expect a specific score value, and `assertScore(Interval, ...)` method when we'd like to check if a score value belongs to a range.

Reply:
@sourabhsparkala

Reply:
I think this is an open discussion. As this is still not concrete. Shall we discuss this another time? For now, precision 0.001 is a really good benchmark.

Reply:
Sure.